gs_switcher.lua

```diff
@@ -125,7 +125,8 @@ hook_Add("PlayerBindPress", "GS_WeaponSelector", function(pPlayer, sBind, bPress
 	if (sBind == "lastinv") then
 		if (bPressed) then
 			local pLastWeapon = pPlayer:GetInternalVariable("m_hLastWeapon")
-			if IsValid(pLastWeapon) then
+
+			if (pLastWeapon ~= nil and pLastWeapon:IsWeapon()) then
```

Comment:
Good idea. Didn't think that anyone would be evil enough to set it to a non-weapon entity.